R/outbreaker_data.R

```diff
@@ -111,12 +111,15 @@ outbreaker_data <- function(..., data = list(...)) {
     }
 
 
-    ## Remove trailing zeroes to prevent starting with -Inf temporal loglike
-    if(data$w_dens[length(data$w_dens)] < 1e-15) {
-      final_index <- max(which(data$w_dens > 1e-15))
-      data$w_dens <- data$w_dens[1:final_index]
+    ## Replace any 0s with minimum value in w_dens
+    if(any(!is.finite(log(data$w_dens)))){
+      is_positive <- is.finite(log(data$w_dens))
+      to_replace <- !is_positive
+      val_replace <- min(data$w_dens[is_positive])
+      data$w_dens[to_replace] <- val_replace
     }
```

Comment:
I would suggest adding a warning to keep the user informed of the issue. See the warning message: we should add additional info in @details in the documentation of `outbreaker_data`.

Also, the feature provided here may be re-used (as you pointed out, e.g. for `f_dens`) so it would be great to have it implemented as a function e.g. `sanitize_pmf`, in a separate file. 

```suggestion
    ## Replace any 0s with minimum value in w_dens
    if(any(!is.finite(log(data$w_dens)))){
      msg <- paste("Non-positive values in `w_dens` have been replaced with small probabilities.",
                   "See `?outbreaker_data` of details", sep = "\n")
      warning(msg)
      is_positive <- is.finite(log(data$w_dens))
      to_replace <- !is_positive
      val_replace <- min(data$w_dens[is_positive])
      data$w_dens[to_replace] <- val_replace
    }
```